### PR TITLE
Add SCRAM tests

### LINKED ISSUE #681: Add tests for `SCRAM-SHA-1` and `SCRAM-SHA-256`
For each driver test (dotnet, java, python), we should add additional authentication tests that use `SCRAM-SHA-1` (FerretDB/FerretDB#2012) and `SCRAM-SHA-256` (FerretDB/FerretDB#3784) authentication mechanisms. They are expected to pass for MongoDB and fail (for now) for FerretDB.

### PR DESCRIPTION
**Pull request by b1ron**
Closes #681.

~MongoDB is now configured with authorization enabled by default using a [keyfile](https://www.mongodb.com/docs/manual/core/security-internal-authentication/#std-label-internal-auth-keyfile), so that we can have a single environment. And it's easier than generating x.509 certs and using TLS.~

Not enabling authorization yet as it's not needed to verify credentials.

### CODE REVIEW

**Review by AlekSi — COMMENTED** [review]
There is a lot of CI failures

**Comment by b1ron** [comment]
https://github.com/FerretDB/FerretDB/pull/3996 fixes the Java failures.

**Review by AlekSi — COMMENTED** [review]
https://github.com/FerretDB/FerretDB/pull/3996 was merged, CI restarted – and a bunch of tests still fail

File: Taskfile.yaml [hunk 1]
```diff
@@ -103,16 +104,29 @@ tasks:
     desc: "Run mongo shell"
     cmds:
       - >
-        docker compose run --rm mongosh mongosh mongodb://host.docker.internal:27017/
-        --verbose --eval 'disableTelemetry()' --shell
+        docker compose run --rm mongosh mongosh 
+        'mongodb://host.docker.internal:27017/?directConnection=true'
```
[the remark this answers is not in the document]
Comment (b1ron) [reply Taskfile.yaml:108]: That's a remnant from testing that can be removed.

**Comment by b1ron** [comment]
I just noticed a bug that I introduced [here](https://github.com/FerretDB/dance/blob/main/internal/config/config.go#L192-L199) in the comparison switch statement. I will fix it a separate PR.

File: internal/config/config.go [hunk 2]
```diff
@@ -207,11 +207,11 @@ func (tc *TestConfig) Compare(results *TestResults) (*CompareResult, error) {
 		case Skip:
 			switch testRes.Status {
 			case Fail:
-				compareResult.UnexpectedFail[test] = testResOutput
+				compareResult.ExpectedFail[test] = testResOutput
```
Comment (AlekSi) [inline internal/config/config.go:210]: If we expected (`expectedRes`) test to be skipped (`Skip`) and it actually (`testRes.Status`) failed (`Fail`) instead, that's unexpected by definition. That change and the changes below make CI always green by effectively ignoring all test results.

That's one way to fix everything, for sure, but not the best one 😆

Comment (b1ron) [reply internal/config/config.go:210]: This was an unintended commit! I will revert now to at least show that initial bug.

Comment (AlekSi) [reply internal/config/config.go]: > Please send a separate PR with a code fix

So I did that, and now a bunch of configurations fail

**Review by AlekSi — COMMENTED** [review]
https://github.com/FerretDB/dance/pull/738#discussion_r1467274535

File: internal/config/config.go (hunk 2 above)
Comment (b1ron) [reply internal/config/config.go]: Of course we have to fix the config files.

Comment (b1ron) [reply internal/config/config.go]: Once we merge #751 the configuration failures will be fixed.

**Review by AlekSi — COMMENTED** [review]
Please resolve conflicts

**Review by AlekSi — APPROVED** [review]

**Pull request merged**